src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -973,6 +973,8 @@ async Task<(bool, ImmutableArray<ProjectConfiguration>)> CheckAsync(UpToDateChec
 
                 var fileSystemOperations = new FileSystemOperationAggregator(_fileSystem, logger);
 
+                logger.FileSystemOperations = fileSystemOperations;
```

Comment:
Can this be in ctor rather than property assignment?

Reply:
Unfortunately not. I don't like this either. I'm considering quite a large refactor of the FUTDC which would avoid this issue, among other things.